core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -960,7 +960,7 @@ public void onSuccess(JsonResponse<TaskInfo> result)
                     // if cleanup operation has not at least started task termination, mark the task failed
                     TaskState taskState = getTaskInfo().getTaskStatus().getState();
                     if (!taskState.isTerminatingOrDone()) {
-                        fatalUnacknowledgedFailure(new TrinoTransportException(REMOTE_TASK_ERROR, fromUri(request.getUri()), format("Unable to %s task at %s, last known state was: %s", action, request.getUri(), taskState)));
+                        fatalAsyncCleanupFailure(new TrinoTransportException(REMOTE_TASK_ERROR, fromUri(request.getUri()), format("Unable to %s task at %s, last known state was: %s", action, request.getUri(), taskState)));
```

Comment:
should `fatalUnacknowledgedFailure` lose `instanceof TrinoTransportException` now?

Reply:
Technically it's still possible for a fatal unacknowledged failure to be triggered with `TrinoTransportException` before cleanup is attempted, and when that occurs we can transition the `TaskInfo` to failed immediately which should be preferable (cleanup will still be attempted, but we're not anticipating that succeeding and we don't need to wait for it to fail before reflecting that in `TaskInfo`).